esphome/components/light/automation.h

```diff
@@ -77,16 +79,29 @@ template<typename... Ts> class DimRelativeAction : public Action<Ts...> {
     float rel = this->relative_brightness_.value(x...);
     float cur;
     this->parent_->remote_values.as_brightness(&cur);
-    float new_brightness = clamp(cur + rel, 0.0f, 1.0f);
+    float new_brightness = clamp(cur + rel, min_brightness_, max_brightness_);
+    if ((limit_mode_ == LimitMode::DO_NOTHING) && ((cur < min_brightness_) || (cur > max_brightness_))) {
+      return;
+    }
```

Comment:
No point doing the clamping if the value was out of range to begin
```suggestion
    if ((limit_mode_ == LimitMode::DO_NOTHING) && ((cur < min_brightness_) || (cur > max_brightness_))) {
      return;
    }
    float new_brightness = clamp(cur + rel, min_brightness_, max_brightness_);
```